lib/rubocop/cop/rspec/factory_bot/association_style.rb

```diff
@@ -0,0 +1,254 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module FactoryBot
+        # Use consistent style to define associations.
+        #
+        # @safety
+        #   This cop may cause false-positives in `EnforcedStyle: explicit`
+        #   case. It recognizes any method call that has no arguments as an
+        #   implicit association but it might be a user-defined trait call.
+        #
+        # @example EnforcedStyle: implicit (default)
+        #   # bad
+        #   association :user
+        #
+        #   # good
+        #   user
+        #
+        # @example EnforcedStyle: explicit
+        #   # bad
+        #   user
+        #
+        #   # good
+        #   association :user
+        #
+        #   # good (defined in NonImplicitAssociationMethodNames)
+        #   skip_create
+        class AssociationStyle < Base # rubocop:disable Metrics/ClassLength
+          extend AutoCorrector
+
+          include ConfigurableEnforcedStyle
+
+          DEFAULT_NON_IMPLICIT_ASSOCIATION_METHOD_NAMES = %w[
+            association
+            sequence
+            skip_create
+            traits_for_enum
+          ].freeze
+
+          RESTRICT_ON_SEND = %i[factory trait].freeze
+
+          # @param node [RuboCop::AST::SendNode]
+          # @return [void]
+          def on_send(node)
+            bad_associations_in(node).each do |association|
+              add_offense(
+                association,
+                message: "Use #{style} style to define associations."
+              ) do |corrector|
+                autocorrect(corrector, association)
+              end
+            end
+          end
+
+          private
+
+          # @!method explicit_association?(node)
+          #   @param node [RuboCop::AST::SendNode]
+          #   @return [Boolean]
```

Comment:
I think it returns a `Node`

Reply:
In fact, `@return [true, nil]` seems to be the exact value. Given a pattern with no captures, `def_node_matcher` generates a method which will return `true` or `nil` (when called without block).

- https://github.com/rubocop/rubocop-ast/blob/e487124ee4a5160fd85d2a47ee4adde6ebcd0775/lib/rubocop/ast/node_pattern/method_definer.rb#L12-L15

The real example would look like this:

```ruby
def explicit_association?(param0 = self)
  node = param0
  
  return unless node.is_a?(::RuboCop::AST::Node) && node.children.size  + 1 >= 4 &&
node.send_type? &&
node.children[0].nil?() &&
:association === node.children[1] &&
node.children[2].is_a?(::RuboCop::AST::Node) && node.children[2].sym_type? &&
true
block_given? ? yield() : (return true)

end
```

And since this method is intended to be used to return a boolean, I think `@return [Boolean]` would be acceptable.
Or perhaps the policy could be to always forbid `@param` and `@return` to be written in YARD, as is taken in rubocop-rails.